src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -87,3 +119,303 @@ export const validateStorageSubmissionParams = celebrate({
     version: Joi.number().required(),
   }),
 })
+
+export const createFormsgReqBody: SharedSubmissionMiddlewareHandlerType =
+  async (req, res, next) => {
+    if (req.body.formsg) return res.send(new FormsgReqBodyExistsError())
+    else {
+      req.body.formsg = {}
+      return next()
+    }
+  }
+
+export const retrieveForm: SharedSubmissionMiddlewareHandlerType = async (
+  req,
+  res,
+  next,
+) => {
+  const { formId } = req.params
+
+  const logMeta = {
+    action: 'retrieveForm',
+    ...createReqMeta(req),
+    formId,
+  }
+
+  // Retrieve form
+  const formResult = await FormService.retrieveFullFormById(formId)
+  if (formResult.isErr()) {
+    logger.warn({
+      message: 'Failed to retrieve form from database',
+      meta: logMeta,
+      error: formResult.error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(formResult.error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  req.body.formsg.formDef = formResult.value
+  return next()
+}
+
+export const checkPublicKey: SharedSubmissionMiddlewareHandlerType = async (
+  req,
+  res,
+  next,
+) => {
+  const formDef = req.body.formsg.formDef
+
+  if (!formDef) return res.send(new FormDefinitionNotRetrievedError())
```

Comment:
at this point, it should no longer be undefined, i.e. formDef it should be part of the function parameter signature so that typescript can check that it is correctly passed on

it does mean also correctly setting the type for the preceding middleware :)

Reply:
Addressed in [e081264](https://github.com/opengovsg/FormSG/pull/6595/commits/e081264c64aa0ebb92069bba5f1a7e8c5461c91d), resolving to streamline discussions [here](https://github.com/opengovsg/FormSG/pull/6595#discussion_r1298348792).